cmd/check-spelling/README.md

```diff
@@ -0,0 +1,176 @@
+```
+
+## Other options
+
+Lists all available options and commands:
+
+```sh
+$ ./kata-spell-check.sh -h
+```
+
+## Technical details
+
+### Hunspell dictionary format
+
+A Hunspell dictionary comprises two text files:
+
+- A word list file
+
+  A list of words, one per line with optional references to one or more
+  rules defined in the rules file and optional comments.
+
+  Fixed words (such as company names) are specified verbatim. However,
+  "normal" words should be entered in their *root form*.
+
+  A *root* form for a word is simply the simplest and shortest form of a
+  word. For example, the words "computer", "computers", "computer's",
+  "computing" and "computed" can all be formed from the root word which is
+  "computer". All the other words can be constructed from the word
+  "computer" by a combination of applying the following manipulations:
+
+  - Removing one or more characters from the end of the word.
+  - Adding a new ending.
+
+- A rules file
+
+  This file defines named manipulations that are applied to root words to
+  form new words. For example, there are rules to make a root word plural.
+
+### Source files
+
+The rules file and word list file for the custom dictionary are
+generated from "source" fragment files in the [`data`](data/) directory.
+
+All the fragment files allow comments using the hash (`#`) comment
+symbol and all files contain a comment header explaining their content.
+
+#### Word list file fragments
+
+The `*.txt` files are word list file fragments. Splitting the word list
+into fragments makes updates easier and clearer as each fragment is a
+grouping of related terms. The name of the file gives a clue as to the
+contents but the comments at the top of each file provide further
+detail.
+
+Every line that does not start with a comment symbol contains a single
+word. An optional comment for a word may appear after the word and is
+separated from the word by whitespace followed by the comment symbol:
+
+```
+word		# This is a comment explaining this particular word list entry.
+```
+
+Each word *may* also be suffixed by a forward slash followed by one or more
+upper-case letters. Each letter refers to a rule name in the rules file:
+
+```
+word/AC		# This word references the 'A' and 'C' rules.
+```
+
+#### Rules file
+
+The [rules file](data/rules.aff) contains a set of general rules that can be
+applied to one or more root words in the word list files. Comments are
+permitted in the rules file.
+
+For an explanation of the format of this file see
+[`man 5 hunspell`](http://www.manpagez.com/man/5/hunspell)
+([source](https://github.com/hunspell/hunspell/blob/master/man/hunspell.5)).
+
+## Adding a new word
+
+### Update the word list fragment
+
+If you wish to allow a new word to the dictionary,
+
+- Check to ensure the word really does need to be added
+
+  Is the word actually valid and correct? If it is a project / product /
+  company name is the capitalization correct?
```

Comment:
Lines 134-139 suggested rewrite: 
If you want to allow a new word to the dictionary, 
- Check to ensure you do need to add the word 
Is the word valid and correct? If the word is a project, product, or company name, is the capitalization correct?